Source/WebKit/Shared/RemoteLayerTree/RemoteLayerTreePropertyApplier.mm

```diff
@@ -173,6 +173,79 @@ static MTCoreMaterialRecipe materialRecipeForAppleVisualEffect(AppleVisualEffect
     }
 }
 
+static MTCoreMaterialVisualStyle materialVisualStyleForAppleVisualEffect(AppleVisualEffect effect)
+{
+    switch (effect) {
+    case AppleVisualEffect::VibrancyLabel:
+    case AppleVisualEffect::VibrancyFill:
+        return PAL::get_CoreMaterial_MTCoreMaterialVisualStylePrimary();
+    case AppleVisualEffect::VibrancySecondaryLabel:
+    case AppleVisualEffect::VibrancySecondaryFill:
+        return PAL::get_CoreMaterial_MTCoreMaterialVisualStyleSecondary();
+    case AppleVisualEffect::VibrancyTertiaryLabel:
+    case AppleVisualEffect::VibrancyTertiaryFill:
+        return PAL::get_CoreMaterial_MTCoreMaterialVisualStyleTertiary();
+    case AppleVisualEffect::VibrancyQuaternaryLabel:
+        return PAL::get_CoreMaterial_MTCoreMaterialVisualStyleQuaternary();
+    case AppleVisualEffect::VibrancySeparator:
+        return PAL::get_CoreMaterial_MTCoreMaterialVisualStyleSeparator();
+    case AppleVisualEffect::None:
+    case AppleVisualEffect::BlurUltraThinMaterial:
+    case AppleVisualEffect::BlurThinMaterial:
+    case AppleVisualEffect::BlurMaterial:
+    case AppleVisualEffect::BlurThickMaterial:
+    case AppleVisualEffect::BlurChromeMaterial:
+        ASSERT_NOT_REACHED();
+        return nil;
+    }
+}
+
+static MTCoreMaterialVisualStyleCategory materialVisualStyleCategoryForAppleVisualEffect(AppleVisualEffect effect)
+{
+    switch (effect) {
+    case AppleVisualEffect::VibrancyLabel:
+    case AppleVisualEffect::VibrancySecondaryLabel:
+    case AppleVisualEffect::VibrancyTertiaryLabel:
+    case AppleVisualEffect::VibrancyQuaternaryLabel:
+        return PAL::get_CoreMaterial_MTCoreMaterialVisualStyleCategoryStroke();
+    case AppleVisualEffect::VibrancyFill:
+    case AppleVisualEffect::VibrancySecondaryFill:
+    case AppleVisualEffect::VibrancyTertiaryFill:
+    case AppleVisualEffect::VibrancySeparator:
+        return PAL::get_CoreMaterial_MTCoreMaterialVisualStyleCategoryFill();
+    case AppleVisualEffect::None:
+    case AppleVisualEffect::BlurUltraThinMaterial:
+    case AppleVisualEffect::BlurThinMaterial:
+    case AppleVisualEffect::BlurMaterial:
+    case AppleVisualEffect::BlurThickMaterial:
+    case AppleVisualEffect::BlurChromeMaterial:
+        ASSERT_NOT_REACHED();
+        return nil;
+    }
+}
+
+static void applyVisualStylingToLayer(CALayer *layer, AppleVisualEffect material, AppleVisualEffect visualStyling)
+{
+    MTCoreMaterialRecipe recipe = PAL::get_CoreMaterial_MTCoreMaterialRecipeNone();
+    if (material != AppleVisualEffect::None)
+        recipe = materialRecipeForAppleVisualEffect(material);
```

Comment:
Nit: would it be wrong to fold in `PAL::get_CoreMaterial_MTCoreMaterialRecipeNone()` as the return value for `AppleVisualEffect::None`?

Reply:
Done. It should be fine – not sure why I didn't do that in the first place 😅